modules/library/user-favorites.php

```diff
@@ -0,0 +1,122 @@
+<?php
+namespace Elementor\Modules\Library;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class User_Favorites {
+	const USER_META_KEY = 'elementor_library_favorites';
+
+	/**
+	 * @var int
+	 */
+	private $user_id;
+
+	/**
+	 * @var array|null
+	 */
+	private $cache;
+
+	/**
+	 * User_Favorites constructor.
+	 *
+	 * @param $user_id
+	 */
+	public function __construct( $user_id ) {
+		$this->user_id = $user_id;
+	}
+
+	/**
+	 * @param null  $vendor
+	 * @param null  $resource
+	 * @param false $ignore_cache
+	 *
+	 * @return array
+	 */
+	public function get( $vendor = null, $resource = null, $ignore_cache = false ) {
+		if ( $ignore_cache || empty( $this->cache ) ) {
+			$this->cache = get_user_meta( $this->user_id, self::USER_META_KEY, true );
+		}
+
+		if ( ! $this->cache || ! is_array( $this->cache ) ) {
+			return [];
+		}
+
+		if ( $vendor && $resource ) {
+			$key = $this->get_key( $vendor, $resource );
+
+			return isset( $this->cache[ $key ] ) ? $this->cache[ $key ] : [];
+		}
+
+		return $this->cache;
+	}
+
+	/**
+	 * @param       $vendor
+	 * @param       $resource
+	 * @param array $value
+	 *
+	 * @return $this
+	 */
+	public function save( $vendor, $resource, $value = [] ) {
+		$all_favorites = $this->get();
+
+		$all_favorites[ $this->get_key( $vendor, $resource ) ] = $value;
+
+		update_user_meta( $this->user_id, self::USER_META_KEY, $all_favorites );
```

Comment:
```suggestion
		$success = update_user_meta( $this->user_id, self::USER_META_KEY, $all_favorites );
		
		if ( $success ) ...
```
Here and all methods should return a bool (success/failed)